src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -149,13 +150,16 @@ public function onSubmit(FormEvent $event)
         }
 
         if ($this->deleteEmpty) {
-            $previousData = $event->getForm()->getData();
+            $previousData = $form->getData();
+            $entryFilter = is_callable($this->deleteEmpty) ? $this->deleteEmpty : function () {return false; };
             foreach ($form as $name => $child) {
+                /* @var $child FormInterface */
                 $isNew = !isset($previousData[$name]);
+                $isEmpty = $child->isEmpty() || true === $entryFilter($child->getData());
```

Comment:
The `$entryFilter` and the empty closure function looks a bit overkill, can be simplified to:
```
$isEmpty = $child->isEmpty() || (is_callable($this->deleteEmpty) && true === $this->deleteEmpty($child->getData());
```